### PR TITLE
*: Improve go build

### PR DESCRIPTION
**Pull request by fanminshi**
remove go install from build scripts.
go build for operators can now take-in extra flags.
refactor go_build as a standalone lib in hack/lib.

### CODE REVIEW

**Comment by fanminshi** [comment]
cc/ @hongchaodeng

File: hack/test [hunk 1]
```diff
@@ -3,6 +3,7 @@
 set -o errexit
 set -o nounset
 set -o pipefail
+set -x
```
Comment (hongchaodeng) [inline hack/test]: Remove this?

File: hack/build/backup-operator/build [hunk 2]
```diff
@@ -6,13 +6,7 @@ set -o pipefail
 
 function go_build {
```
Comment (hongchaodeng) [inline hack/build/backup-operator/build]: Can you put this into a lib?

File: hack/test [hunk 3]
```diff
@@ -95,9 +96,10 @@ function build_restore_operator  {
 }
 
 function build_pass {
-	IMAGE=$OPERATOR_IMAGE hack/build/operator/build
-	build_backup_operator
-	build_restore_operator
+	# cache all builds by passing '-i'. 
+	IMAGE=$OPERATOR_IMAGE hack/build/operator/build -i
```
Comment (hongchaodeng) [inline hack/test]: Don't enable install in CI.

Comment (fanminshi) [reply hack/test]: should we enable cache in CI so that is will be faster to run? I thought that's the purpose of the support caching for go build.

Comment (hongchaodeng) [reply hack/test]: >  I thought that's the purpose of the support caching for go build.

No. We add it for local dev purpose.

**Comment by fanminshi** [comment]
PTAL cc/ @hongchaodeng

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**